tests/unit/test_download_utilities.py

```diff
@@ -74,3 +76,22 @@ def test_download_granules_failure(self):
         self.assertTrue(str(context_manager.exception).startswith(
             'Download failed: Unable to download a url of unknown type'
         ))
+
+    @patch('harmony_netcdf_to_zarr.download_utilities.Process')
+    def test_download_granules_process_error(self, mock_process):
+        """Check that a request to download all files that experiences an
+        error exit raises an expected assertion .
+
+        """
+        # Convert output to set, as order may not be preserved:
+        processes = [Mock(Process), Mock(Process), Mock(Process)]
+        for p in processes:
+            p.exitcode = 0
+        processes[0].exitcode = -9
+        mock_process.side_effect = processes
+
+        regexMessage = 'Error Exit occurred downloading data to Harmony: processes exit codes: \[-9, 0, 0\]'
```

Comment:
Super OCD nit: Python variable names are `snake_case`. (It's not really that important here, but I spotted it and couldn't help myself)